rekor_common.proto

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v2.common;
+
+option go_package = "github.com/sigstore/rekor/pkg/generated/protobuf/common";
+option java_package = "dev.sigstore.rekor.v2.common";
+option java_multiple_files = true;
+option java_outer_classname = "RekorCommonProto";
+
+message Signature {
+    /*
+     * The crypto algorithm used to generate the signature
+     */
+    PKIAlgorithm algorithm = 1;
+    /*
+     * The value of the digital signature
+     */
+    bytes content          = 2;
+    /*
+     * The public key that can be used to verify the signature
+     */
+    PublicKey public_key   = 3;
+}
+
+enum PKIAlgorithm {
+    PKI_ALGORITHM_UNSPECIFIED = 0;
+    PGP                       = 1;
+    MINISIGN                  = 2;
+    X509                      = 3;
+    SSH                       = 4;
+    PKCS7                     = 5;
+    TUF                       = 6;
+}
+
+message PublicKey {
+    /*
+     * The public key
+     */
+    bytes content = 1;
```

Comment:
This can contain either a public key, certificate or chain, correct? Should we rename this?

Reply:
well, it could contain a PGP public key, an X509 certificate (or certificate chain), a TUF metadata root, etc. We could use a oneOf with specific types here if you think its better

Reply:
I'd be in support of switching to a oneOf. I think it makes it more explicit and less prone to error.

Reply:
FYI this linked issue https://github.com/sigstore/rekor/issues/809 -- would support switching to a oneof